### ISSUE TITLE
more dtrace probes: glob, hash

### ISSUE THREAD

**Issue by rurban** [perl11/cperl#240]
for internal profiling: glob, hash (entry+return)

glob-mode
```
       0: gv_init
       1: gv_add
       2: gv_fetch
       3: gv_fetchmeth
```
(autoload?, stash?)

hash-mode 
```
       0: hv_fetch
       1: hv_store
       2: hv_exists
       3: hv_delete
```
(magical?)

# glob

 - sudo dtrace -qn "BEGIN{self->ts=0}glob-entry{self->ts=timestamp}glob-return{ts=timestamp-self->ts;@a[arg0]=avg(ts);@q[arg0]=quantize(ts);@c[arg0]=count()}" -c"./miniperl -Ilib autodoc.pl"

average time per mode (ns):

                1             1234  # ADD
                0             1417  # INIT
                2             3055  # FETCH
                3             4150  # FETCHMETH

time spent per mode (ns):

                3
           value  ------------- Distribution ------------- count
            1024 |                                         0
            2048 |@@@@@@@@@@@@@                            1
            4096 |@@@@@@@@@@@@@@@@@@@@@@@@@@@              2
            8192 |                                         0

                1
           value  ------------- Distribution ------------- count
             256 |                                         0
             512 |@@@@@@@@@@@@@@@@@@@@@@@@@                67
            1024 |@@@@@@@@@@@                              31
            2048 |@@@                                      8
            4096 |@                                        2
            8192 |                                         0

                0
           value  ------------- Distribution ------------- count
             256 |                                         0
             512 |@                                        8
            1024 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@     273
            2048 |@@                                       16
            4096 |                                         3
            8192 |                                         0

                2
           value  ------------- Distribution ------------- count
            1024 |                                         0
            2048 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@    224
            4096 |@@                                       13
            8192 |                                         3
           16384 |                                         1
           32768 |                                         0

sum of glob calls:

                3                3  # FETCHMETH
                1              108  # ADD
                2              241  # FETCH
                0              300  # INIT

# hash

sum of hash calls:

 - sudo dtrace -n "hash-entry{@[arg0]=count()}" -c"./miniperl -Ilib script.pl"

                3               15  # DELETE
                2               39  # EXISTS
                1              125  # STORE
                0              724  # FETCH

Time spent (on average):

 - sudo dtrace -qn "BEGIN{self->ts=0}hash-entry{self->ts=timestamp}hash-return{@[arg0]=avg(timestamp-self->ts)}" -c"./miniperl -Ilib script.pl"

                0             1000  # FETCH
                2             1014  # EXISTS
                1             1111  # STORE
                3             1641  # DELETE

**Comment by rurban** [comment]
ready, but not merged. unsure about performance impact. benchmarks not conclusive.

**Comment by rurban** [comment]
Merged with 2a1b0fdf6e73934a4d4fc91bf1a2063100991213 to 5.27.0
